README.md

```diff
@@ -112,6 +112,37 @@ Notifies about new events matching the filter provided via webhooks.
 > | filter  |  required | JSON object ([nostr.Filter](https://pkg.go.dev/github.com/nbd-wtf/go-nostr@v0.25.7#Filter)) | Filters to subscribe to |
 
 
+#### Response
+
+> ```json
+> {
+>   "subscription_id": "f370d1fc-x0x0-x0x0-x0x0-8f68fa12f32c",
+>   "webhookUrl": "https://your.webhook.url"
+> }
+>```
+</details>
+
+------------------------------------------------------------------------------------------
+
+### Subscribe to NIP-47 Events
+
+Notifies about new NIP-47 response events which are requested by the pubkey to the wallet service.
+
+<details>
+<summary>
+<code>POST</code> <code><b>/nip47/subscriptions</b></code>
+</summary>
+
+#### Request Body
+
+> | name      |  type     | data type               | description                                                           |
+> |-----------|-----------|-------------------------|-----------------------------------------------------------------------|
+> | relayUrl  |  optional | string           | If no relay is provided, it uses the default relay  |
+> | webhookUrl  |  required | string         | Webhook URL to publish events |
+> | walletPubkey  |  optional | string         | Public key of the wallet service |
+> | pubkey  |  optional | string         | Public key of the user |
```

Comment:
I think these should not be optional, right?

Reply:
😄 I didn't know what to do here, it's actually either-or for pubkey and wallet pubkey. Even without specifying both we can still listen to response kind events on the relay, but yeah ideally we should provide both.

Reply:
However if you don't (especially if you do not provide the `pubkey`), http-nostr will send extra events which cannot be decrypted with the subscriber's private key and will probably cause issues with the subscribing application. I think it actually makes sense to require them both to ensure the app is subscribing to what it thinks it's subscribing.

Reply:
you miss that if you build a wallet then you specify only the walletPubkey. 

what can be subscribed should ultimately be up for some authentication. e.g. proof you are the wallet if you only subscribe to the walletPubkey
and proof you are the pubkey if you add the pubkey.

Reply:
but this either/or should be reflected in the docs. simply marking them both as optional is wrong.

Reply:
Ah, I see. I thought this was for the new NIP-47 notifications which are always sent in one direction, not for standard NIP-47 request/responses

Reply:
@bumi if this is for notifications then I would say all are required, or if not, should notifications be a separate API endpoint so that the requirements and responses are clearly separated from this one?

Reply:
@rolznz what do you mean? 

for me this API was either for apps that want to subscribe to relevant notifications or for wallets. 
but you're right, lets make it only for apps and we require a 
wallet pubkey and a connection pubkey.

Reply:
Cool, so now we require both connection pubkey and wallet pubkey, and keep listening to all the response events the user is receiving from the NWC wallet provider

That means:
- This is not related to `23196` kind/NIP-47 notifications
- The subscriptions only post the responses to the webhook, the user has to publish request events themselves (using our other endpoints, ideally)